pkg/loadbalancer/experimental/benchmark_test.go

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package experimental_test
+
+import (
+	"encoding/binary"
+	"fmt"
+	"testing"
+
+	"github.com/cilium/cilium/pkg/clustermesh/types"
+	"github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/loadbalancer/experimental"
+	"github.com/cilium/cilium/pkg/source"
+)
+
+func BenchmarkInsertService(b *testing.B) {
+	p := servicesFixture(b)
+
+	b.ResetTimer()
+
+	numObjects := 1000
+
+	// Add 'numObjects' existing objects to the table.
+	wtxn := p.Services.WriteTxn()
+	for i := 0; i < numObjects; i++ {
+		name := loadbalancer.ServiceName{Namespace: "test-existing", Name: fmt.Sprintf("svc-%d", i)}
+		var addr1 [4]byte
+		binary.BigEndian.PutUint32(addr1[:], 0x02000000+uint32(i))
+		addrCluster, _ := types.AddrClusterFromIP(addr1[:])
+		p.Services.UpsertFrontend(
+			wtxn,
+			&experimental.FrontendParams{
+				Name:    name,
+				Address: *loadbalancer.NewL3n4Addr(loadbalancer.TCP, addrCluster, 12345, loadbalancer.ScopeExternal),
+				Type:    loadbalancer.SVCTypeClusterIP,
+				Source:  source.Kubernetes,
+			},
+		)
+	}
+	wtxn.Commit()
+
+	// Benchmark the speed at which a new service is upserted. 'numObjects' are inserted in one
```

Comment:
Nit: The comment is a bit confusing: large `numObjects` amortises `p.Services.WriteTxn()` from line 46 and `wtxn.Abort()` from line 62 (Abort, not Commit). However, `b.N` should indeed amortise `p.Services.WriteTxn()` and `wtxn.Commit()` from lines 25 and 41.

Reply:
Oh yeah indeed confusing. Fixed. The benchmark itself is doing what's intended: against 1k existing services & frontends, insert 1k new ones repeatedly (throwing away the result, so repeating the same operation over and over).